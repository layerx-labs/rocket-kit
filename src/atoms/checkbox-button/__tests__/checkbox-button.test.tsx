import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { CheckboxButton } from '../../..';

describe('CheckboxButton', () => {
  it('renders', () => {
    const { asFragment } = render(
      <CheckboxButton value={'check'} label={'hello'} checked={true} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('is checked', () => {
    render(<CheckboxButton value={'check'} label={'hello'} checked={true} />);
    expect(screen.getByRole('checkbox')).toBeChecked();
  });

  it('is not checked', () => {
    render(<CheckboxButton value={'check'} label={'hello'} checked={false} />);
    expect(screen.getByRole('checkbox')).not.toBeChecked();
  });

  it('is not checked by default', () => {
    render(<CheckboxButton value={'check'} label={'hello'} />);
    expect(screen.getByRole('checkbox')).not.toBeChecked();
  });

  it('is changes to checked when clicked', () => {
    const onChangeMock = jest.fn();
    render(
      <CheckboxButton value={'check'} label={'hello'} onChange={onChangeMock} />
    );
    const checkboxElem = screen.getByRole('checkbox');
    userEvent.click(checkboxElem);
    expect(onChangeMock).toBeCalledTimes(1);
    expect(checkboxElem).toBeChecked();
  });
});
