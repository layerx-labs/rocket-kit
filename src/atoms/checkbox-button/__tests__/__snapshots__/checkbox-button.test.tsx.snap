// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxButton renders 1`] = `
<DocumentFragment>
  .c0 {
  border: 0.1875rem solid #2ba8b6;
  border-radius: 0.375rem;
  background-color: #56cad7;
  height: 3.125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c0:hover input:not(:disabled) ~ span {
  border-color: #939393;
}

.c0:hover {
  background-color: #56cad7;
}

.c0 span {
  color: #ffffff;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c1 {
  margin-left: 2.5rem;
  line-height: 1.5rem;
}

.c2 {
  position: absolute;
  left: 0;
  opacity: 0;
  cursor: pointer;
}

.c2:checked:not(:disabled) ~ span {
  background-color: #ffffff;
  border-color: #ffffff;
}

.c2:checked:not(:disabled) ~ span:after {
  display: block;
}

.c2:checked:disabled ~ span {
  border-color: transparent;
  background-color: #939393;
  color: #939393;
}

.c2:checked:disabled ~ span:after {
  display: block;
}

.c2:not(:checked):disabled ~ span {
  border-color: #939393;
  background-color: #f4f4f4;
}

.c2:hover:not(:disabled):checked ~ span {
  border-color: #ffffff;
}

.c3 {
  position: absolute;
  left: 0.625rem;
  border: 0.125rem solid #939393;
  border-radius: 999px;
  background-color: #ffffff;
  width: 1.5rem;
  height: 1.5rem;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c3:after {
  content: '';
  position: absolute;
  display: none;
  top: 0.1875rem;
  left: 0.375rem;
  width: 0.3125rem;
  height: 0.5625rem;
  border: solid #56cad7;
  border-width: 0 0.1875rem 0.1875rem 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

<label
    class="c0 checkbox-button"
  >
    <span
      class="c1"
    >
      hello
    </span>
    <input
      aria-checked="true"
      checked=""
      class="c2"
      name="check"
      type="checkbox"
    />
    <span
      class="c3"
    />
  </label>
</DocumentFragment>
`;
